src/Features/Core/Portable/UnusedReferences/UnusedReferencesService.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.UnusedReferences
+{
+    [ExportWorkspaceService(typeof(IUnusedReferencesService)), Shared]
+    internal class UnusedReferencesService : IUnusedReferencesService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public UnusedReferencesService()
+        {
+        }
+
+        public async Task<ImmutableArray<Reference>> GetUnusedReferencesAsync(Project project, CancellationToken cancellationToken)
+        {
```

Comment:
if this is real code, check first if project.SupportsCompilation

Reply:
removing this class for now, but good to know for the future.